drivers/phoenixcontact_modbus.c

```diff
@@ -0,0 +1,200 @@
+/*  phoenixcontact_modbus.c - Driver for PhoenixContact-QUINT UPS 
+ *
+ *  Copyright (C)
+ *    2017  Spiros Ioannou <sivann@inaccess.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ */
+
+#include "main.h"
+#include <modbus.h>
+
+#define DRIVER_NAME	"NUT PhoenixContact Modbus driver"
+#define DRIVER_VERSION	"0.01"
+
+#define CHECK_BIT(var,pos) ((var) & (1<<(pos)))
+
+/* Variables */
+modbus_t *ctx = NULL;
+int errcount = 0;
+
+int mrir(modbus_t *ctx, int addr, int nb, uint16_t *dest) ;
+
+/* driver description structure */
+upsdrv_info_t upsdrv_info = {
+	DRIVER_NAME,
+	DRIVER_VERSION,
+	"Spiros Ioannou <sivann@inaccess.com>\n",
+	DRV_EXPERIMENTAL,
+	{ NULL }
+};
+
+void upsdrv_initinfo(void)
+{
+	upsdebugx(2, "upsdrv_initinfo");
+
+	dstate_setinfo("device.mfr", "Phoenix Contact");
+	dstate_setinfo("device.model", "QUINT-UPS 2320461");
```

Comment:
Do we know this is the only supported model? Is it possible to read a Modbus register to determine this?

If the model number is needed in status displays, it can always be set with `override.device.model` in `ups.conf`.